python/lsst/ts/gis/csc.py

```diff
@@ -53,30 +70,50 @@ def __init__(
     async def telemetry_loop(self):
         """Implement the telemetry feed for the GIS."""
         while True:
-            await self.component.update_status()
-            await asyncio.sleep(self.telemetry_interval)
+            try:
+                await self.component.update_status()
+                await asyncio.sleep(self.telemetry_interval)
+            except Exception:
+                self.log.exception("Telemetry loop failed.")
+                raise
 
     @property
     def connected(self):
         """Return whether the component is connected."""
         return self.component.connected
 
     async def handle_summary_state(self):
+        """Handle summary state transitions.
+
+        * Connect to the client if not connected.
```

Comment:
Great that you added this. However, this is only the case for when transiting to DISABLED or ENABLED state. In all other cases the connection with the client is dropped, the mock server stopped and the telemetry loop stopped, so please add that as well.